docs/modules/ROOT/pages/index.adoc

```diff
@@ -38,4 +64,308 @@ include::includes/quarkus-solace-extension-outgoing.adoc[leveloffset=+1, opts=op
+----
+quarkus.solace.host=tcp://localhost:55555
+quarkus.solace.vpn=default
+quarkus.solace.authentication.basic.username=basic
+quarkus.solace.authentication.basic.password=basic
+
+mp.messaging.incoming.temperatures.connector=quarkus-solace
+mp.messaging.incoming.temperatures.consumer.queue.missing-resource-creation-strategy=create-on-start
+mp.messaging.incoming.temperatures.consumer.queue.add-additional-subscriptions=true
+mp.messaging.incoming.temperatures.consumer.queue.subscriptions=hello/foobar
+----
+
+1. When running in dev mode or tests dev services will automatically start a Solace PubSub+ broker and if broker configuration details are not provided the extension automatically picks up the details of broker started by dev services.
+
+2. If `consumer.queue.name` property is not specified, channel name will be used as queue name.
+
+[[receiving-messages-from-solace]]
+== Receiving messages from Solace
+
+Using the previous configuration, Quarkus application can receive message in several possible ways.
+
+__Direct Payload__
+[source,java]
+----
+import org.eclipse.microprofile.reactive.messaging.Incoming;
+import jakarta.enterprise.context.ApplicationScoped;
+@ApplicationScoped
+public class TemperaturesConsumer {
+    @Incoming("temperatures")
+    public void consume(Double temperature) {
+        // process.
+    }
+}
+----
+
+__Message__
+[source,java]
+----
+@ApplicationScoped
+public class TemperaturesConsumer {
+    @Incoming("temperatures")
+    public CompletionStage<Void> consume(Message<Double> msg) {
+        // access record metadata
+        SolaceInboundMetadata metadata = msg.getMetadata(SolaceInboundMetadata.class).orElseThrow();
+        // process the message payload.
+        Double temperature = msg.getPayload();
+        // Acknowledge the incoming message
+        return msg.ack();
+    }
+}
+----
+
+__SolaceInboundMessage__ This is a wrapper to incoming Inbound Message from Solace Messaging API
+[source,java]
+----
+@ApplicationScoped
+public class TemperaturesConsumer {
+    @Incoming("temperatures")
+    public void consume(SolaceInboundMessage<Double> solaceInboundMessage) {
+        // get actual inbound message
+        InboundMessage inboundMessage = solaceInboundMessage.getMessage();
+        // process the message payload.
+        Double temperature = solaceInboundMessage.getPayload();
+        // access record metadata
+        SolaceInboundMetadata metadata = solaceInboundMessage.getMetadata();
+        // ...
+        solaceInboundMessage.ack();
+    }
+}
+----
+
+[[acknowledgement-handling]]
+== Acknowledgment Handling
+
+By default, acknowledgement strategy is set to client acknowledgement. This gives greater control over acknowledgement and ensures that messages are acknowledged only after successful processing.
+
+[source,java]
+----
+@ApplicationScoped
+public class TemperaturesConsumer {
+    @Incoming("temperatures")
+    public CompletionStage<Void> consume(SolaceInboundMessage<Double> msg) {
```

Comment:
Same here, we should switch SolaceInboundMessage with Message. 
And again do we support double payload type ?

Reply:
Same here. I have noticed it lately. I will correct it.